### ISSUE TITLE
installer failing too many colons in address

### ISSUE THREAD

**Issue by austincunningham** [aerogear/mobile-core#135]
## Description
Cluster up fails running ./installer/install.sh fails with the following issue

```bash
TASK [oc-cluster-up : Cluster up] ***********************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "cmd": "oc cluster up --service-catalog=true --host-data-dir=/home/austincunningham/repos/mobile-core/ui/openshift-data --routing-suffix=addr:10.201.82.185.nip.io --public-hostname=addr:10.201.82.185 --version=v3.9.0 --image=openshift/origin", "delta": "0:00:05.358157", "end": "2018-07-12 10:13:03.961671", "msg": "non-zero return code", "rc": 1, "start": "2018-07-12 10:12:58.603514", "stderr": "FAIL\n   Error: could not create OpenShift configuration\n   Caused By:\n     Error: Docker run error rc=1\n     Details:\n       Image: openshift/origin:v3.9.0\n       Entrypoint: []\n       Command: [start --images=openshift/origin-${component}:v3.9.0 --volume-dir=/var/lib/origin/openshift.local.volumes --dns=0.0.0.0:8053 --write-config=/var/lib/origin/openshift.local.config --master=127.0.0.1 --hostname=localhost --public-master=https://addr:10.201.82.185:8443]\n       Error Output:\n         Error: invalid argument \"https://addr:10.201.82.185:8443\" for --public-master=https://addr:10.201.82.185:8443: not a valid host:port: address addr:10.201.82.185:8443: too many colons in address\n\n\n         Usage:\n           openshift start [options]\n\n         Available Commands:\n           etcd                    Launch etcd server\n           master                  Launch a master\n           network                 Launch node network\n           node                    Launch a node\n           template-service-broker Launch a template service broker server\n\n         Options:\n               --certificate-authority=[]: Optional files containing signing authorities to use (in addition to the generated signer) to verify the API server's serving certificate.\n               --cors-allowed-origins=[]: List of allowed origins for CORS, comma separated.  An allowed origin can be a regular expression to support subdomain matching.  CORS is enabled for localhost, 127.0.0.1, and the asset server by default.\n               --create-certs=true: Indicates whether missing certs should be created.\n               --dns='tcp://0.0.0.0:8053': The address to listen for DNS requests on.\n               --etcd='https://0.0.0.0:4001': The address of the etcd server (host, host:port, or URL). If specified, no built-in etcd will be started.\n               --etcd-dir='openshift.local.etcd': The etcd data directory.\n               --expire-days=730: Validity of the certificates in days (defaults to 2 years). WARNING: extending this above default value is highly discouraged.\n               --hostname='lap-rh-austin': The hostname to identify this node with the master.\n               --images='openshift/origin-${component}:${version}': When fetching images used by the cluster for important components, use this format on both master and nodes. The latest release will be used by default.\n               --latest-images=false: If true, attempt to use the latest images for the cluster instead of the latest release.\n               --listen='https://0.0.0.0:8443': The address to listen for connections on (scheme://host:port).\n               --master='https://localhost:8443': The master address for use by OpenShift components (host, host:port, or URL). Scheme and port default to the --listen scheme and port. When unset, attempt to use the first public IPv4 non-loopback address registered on this host.\n               --master-config='': Location of the master configuration file to run from. When running from configuration files, all other command-line arguments are ignored.\n               --network-plugin='': The network plugin to be called for configuring networking for pods.\n               --node-config='': Location of the node configuration file to run from. When running from configuration files, all other command-line arguments are ignored.\n               --pause=false: If true, wait for a signal before starting the controllers.\n               --portal-net='172.30.0.0/16': The CIDR string representing the network that portal/service IPs will be assigned from. This must not overlap with any IP ranges assigned to nodes for pods.\n               --print-ip=false: Print the IP that would be used if no master IP is specified and exit.\n               --public-master='https://localhost:8443': The master address for use by public clients, if different (host, host:port, or URL). Defaults to same as --master.\n               --recursive-resolv-conf='': An optional upstream resolv.conf that will override the DNS config.\n               --signer-expire-days=1825: Validity of the CA certificate in days (defaults to 5 years). WARNING: extending this above default value is highly discouraged.\n               --volume-dir='openshift.local.volumes': The volume storage directory.\n               --write-config='': Directory to write an initial config into.  After writing, exit without starting the server.\n\n         Use \"openshift <command> --help\" for more information about a given command.\n         Use \"openshift options\" for a list of global command-line options (applies to all commands).", "stderr_lines": ["FAIL", "   Error: could not create OpenShift configuration", "   Caused By:", "     Error: Docker run error rc=1", "     Details:", "       Image: openshift/origin:v3.9.0", "       Entrypoint: []", "       Command: [start --images=openshift/origin-${component}:v3.9.0 --volume-dir=/var/lib/origin/openshift.local.volumes --dns=0.0.0.0:8053 --write-config=/var/lib/origin/openshift.local.config --master=127.0.0.1 --hostname=localhost --public-master=https://addr:10.201.82.185:8443]", "       Error Output:", "         Error: invalid argument \"https://addr:10.201.82.185:8443\" for --public-master=https://addr:10.201.82.185:8443: not a valid host:port: address addr:10.201.82.185:8443: too many colons in address", "", "", "         Usage:", "           openshift start [options]", "", "         Available Commands:", "           etcd                    Launch etcd server", "           master                  Launch a master", "           network                 Launch node network", "           node                    Launch a node", "           template-service-broker Launch a template service broker server", "", "         Options:", "               --certificate-authority=[]: Optional files containing signing authorities to use (in addition to the generated signer) to verify the API server's serving certificate.", "               --cors-allowed-origins=[]: List of allowed origins for CORS, comma separated.  An allowed origin can be a regular expression to support subdomain matching.  CORS is enabled for localhost, 127.0.0.1, and the asset server by default.", "               --create-certs=true: Indicates whether missing certs should be created.", "               --dns='tcp://0.0.0.0:8053': The address to listen for DNS requests on.", "               --etcd='https://0.0.0.0:4001': The address of the etcd server (host, host:port, or URL). If specified, no built-in etcd will be started.", "               --etcd-dir='openshift.local.etcd': The etcd data directory.", "               --expire-days=730: Validity of the certificates in days (defaults to 2 years). WARNING: extending this above default value is highly discouraged.", "               --hostname='lap-rh-austin': The hostname to identify this node with the master.", "               --images='openshift/origin-${component}:${version}': When fetching images used by the cluster for important components, use this format on both master and nodes. The latest release will be used by default.", "               --latest-images=false: If true, attempt to use the latest images for the cluster instead of the latest release.", "               --listen='https://0.0.0.0:8443': The address to listen for connections on (scheme://host:port).", "               --master='https://localhost:8443': The master address for use by OpenShift components (host, host:port, or URL). Scheme and port default to the --listen scheme and port. When unset, attempt to use the first public IPv4 non-loopback address registered on this host.", "               --master-config='': Location of the master configuration file to run from. When running from configuration files, all other command-line arguments are ignored.", "               --network-plugin='': The network plugin to be called for configuring networking for pods.", "               --node-config='': Location of the node configuration file to run from. When running from configuration files, all other command-line arguments are ignored.", "               --pause=false: If true, wait for a signal before starting the controllers.", "               --portal-net='172.30.0.0/16': The CIDR string representing the network that portal/service IPs will be assigned from. This must not overlap with any IP ranges assigned to nodes for pods.", "               --print-ip=false: Print the IP that would be used if no master IP is specified and exit.", "               --public-master='https://localhost:8443': The master address for use by public clients, if different (host, host:port, or URL). Defaults to same as --master.", "               --recursive-resolv-conf='': An optional upstream resolv.conf that will override the DNS config.", "               --signer-expire-days=1825: Validity of the CA certificate in days (defaults to 5 years). WARNING: extending this above default value is highly discouraged.", "               --volume-dir='openshift.local.volumes': The volume storage directory.", "               --write-config='': Directory to write an initial config into.  After writing, exit without starting the server.", "", "         Use \"openshift <command> --help\" for more information about a given command.", "         Use \"openshift options\" for a list of global command-line options (applies to all commands)."], "stdout": "WARNING: Binding DNS on port 8053 instead of 53, which may not be resolvable from all clients.\nUsing Docker shared volumes for OpenShift volumes\nUsing 127.0.0.1 as the server IP\nStarting OpenShift using openshift/origin:v3.9.0 ...\n-- Starting OpenShift container ... \n   Creating initial OpenShift configuration", "stdout_lines": ["WARNING: Binding DNS on port 8053 instead of 53, which may not be resolvable from all clients.", "Using Docker shared volumes for OpenShift volumes", "Using 127.0.0.1 as the server IP", "Starting OpenShift using openshift/origin:v3.9.0 ...", "-- Starting OpenShift container ... ", "   Creating initial OpenShift configuration"]}
	to retry, use: --limit @/home/austincunningham/repos/mobile-core/installer/playbook.retry

PLAY RECAP **********************************************************************************************************
localhost                  : ok=13   changed=1    unreachable=0    failed=1   


```
## Environment

*Operating system*:
Linux mint 18
*OpenShift versions*: 
3.9

## Steps to reproduce
Run the installer as outlined at https://docs.aerogear.org/aerogear/latest/getting-started.html#prerequisites

## Extra Notes

The issue is [here](https://github.com/aerogear/mobile-core/blob/95f587ef3a43ef5550ac80364a6fd28a38ebe1e2/installer/install.sh#L358) where the Cluster IP is obtained. In this SO/setup the value returned is with `addr:` in front of it. 

```
$ DEFAULT_CLUSTER_IP=$(ifconfig $(netstat -nr | awk '{if (($1 == "0.0.0.0" || $1 == "default") && $2 != "0.0.0.0" && $2 ~ /[0-9\.]+{4}/){print $NF;} }' | head -n1) | grep 'inet ' | awk '{print $2}')
$ echo $DEFAULT_CLUSTER_IP 
addr:10.32.241.91
```

**Comment by jasonmadigan** [comment]
@austincunningham can you add the output of `ipconfig`? Wondering if we've got some bug with interfaces with IPv6 addresses.

To workaround, use `10.201.82.185` (without the `:`) when prompted.

**Comment by austincunningham** [comment]
```➜  ~ ifconfig
docker0   Link encap:Ethernet  HWaddr 02:42:c3:0e:5f:16  
          inet addr:172.17.0.1  Bcast:172.17.255.255  Mask:255.255.0.0
          inet6 addr: fe80::42:c3ff:fe0e:5f16/64 Scope:Link
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:341391 errors:0 dropped:0 overruns:0 frame:0
          TX packets:419562 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:70291359 (70.2 MB)  TX bytes:42588721 (42.5 MB)

enp0s31f6 Link encap:Ethernet  HWaddr c8:5b:76:bc:1d:61  
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:5266893 errors:3084 dropped:0 overruns:0 frame:2411
          TX packets:2014739 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:5798434996 (5.7 GB)  TX bytes:606306422 (606.3 MB)
          Interrupt:16 Memory:f2200000-f2220000 

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:24504183 errors:0 dropped:0 overruns:0 frame:0
          TX packets:24504183 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:8200711343 (8.2 GB)  TX bytes:8200711343 (8.2 GB)

virbr0    Link encap:Ethernet  HWaddr 00:00:00:00:00:00  
          inet addr:192.168.122.1  Bcast:192.168.122.255  Mask:255.255.255.0
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:230994 errors:0 dropped:0 overruns:0 frame:0
          TX packets:410332 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:9650257 (9.6 MB)  TX bytes:879042367 (879.0 MB)

virbr3    Link encap:Ethernet  HWaddr 00:00:00:00:00:00  
          inet addr:192.168.42.1  Bcast:192.168.42.255  Mask:255.255.255.0
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:9261 errors:0 dropped:0 overruns:0 frame:0
          TX packets:8517 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:4369132 (4.3 MB)  TX bytes:1246391 (1.2 MB)

wlp3s0    Link encap:Ethernet  HWaddr 34:f3:9a:b7:1a:38  
          inet addr:10.201.82.185  Bcast:10.201.82.255  Mask:255.255.255.0
          inet6 addr: fe80::4962:34ce:6280:8adb/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:6407376 errors:0 dropped:0 overruns:0 frame:0
          TX packets:2675379 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:7000950681 (7.0 GB)  TX bytes:830965800 (830.9 MB)

```

**Comment by austincunningham** [comment]
Was on master so , more that likely the issue, manually entering the IP resolves the issue

**Comment by jasonmadigan** [comment]
@austincunningham I wouldn't mind taking a peak at this anyway, just to check master - will re-open and we'll take a look before closing

**Comment by camilamacedo86** [comment]
@austincunningham could you provide the steps/setup to face this issue? Because just following the instructions did not allow us to reproduce it.

**Comment by austincunningham** [comment]
@camilamacedo86 
- follow the getting started guide 
```
git clone https://github.com/aerogear/mobile-core.git
cd mobile-core
```
- don't checkout 1.0.0
- run the installer
```
$ ./installer/install.sh
```
- enter dockerhub credentials
- accept the defaults when prompted,
- error occurs

Suspect it's OS related bug with master

**Comment by camilamacedo86** [comment]
@austincunningham shows by following it that your SO was with some specific setup and it was using 
`addr:10.201.82.185.nip.io` instead of `10.201.82.185.nip.io`.  What shows if you run it now in the cluster IP by default? ( following an example )

<img width="236" alt="screen shot 2018-07-24 at 11 45 29" src="https://user-images.githubusercontent.com/7708031/43133556-14d0cc5c-8f37-11e8-9362-97b5a05fc3db.png">

Could you add the steps performed to work around it?

**Comment by camilamacedo86** [comment]
But what was the valued returned by default? It was returned with the : ?

**Comment by austincunningham** [comment]
Suspect the issue is on this line, https://github.com/aerogear/mobile-core/blob/95f587ef3a43ef5550ac80364a6fd28a38ebe1e2/installer/install.sh#L358 When run locally i get the following 
```
$ DEFAULT_CLUSTER_IP=$(ifconfig $(netstat -nr | awk '{if (($1 == "0.0.0.0" || $1 == "default") && $2 != "0.0.0.0" && $2 ~ /[0-9\.]+{4}/){print $NF;} }' | head -n1) | grep 'inet ' | awk '{print $2}')
$ echo $DEFAULT_CLUSTER_IP 
addr:10.32.241.91
```

**Comment by camilamacedo86** [comment]
Perfect !!! It was what I'd like to confirm :-) The problem here is that the default function is returning the IP with the `addr:` in front of it.

**Comment by wei-lee** [comment]
Hi @austincunningham,

Thanks for your feedback. This repo is now deprecated and is replaced by [Mobile Developer Console](https://github.com/aerogear/mobile-developer-console). Could you please give it a try and if you still experience the same problem, please open a new issue in that project.

Thanks.
